lib/middleman-simple-thumbnailer/image.rb

```diff
@@ -24,22 +25,15 @@ def resized_img_path
     end
 
     def base64_data
+      unless cached_thumbnail_available?
+        resize!
+        save_cached_thumbnail
+      end
       Base64.encode64(File.read(cached_resized_img_abs_path))
     end
 
-    def resize!(resize_to)
-      self.resize_to = resize_to
-      return if cached_thumbnail_available?
-      image.resize(resize_to)
-      save_cached_thumbnail
-    end
-
     def save!
-      if cached_thumbnail_available?
-        FileUtils.cp cached_resized_img_abs_path, resized_img_abs_path
-      else
-        image.write(resized_img_abs_path)
-      end
```

Comment:
Of course, you're right (was there initially, got lost when I rebased on – note to self: even proof of concepts shouldn't be done in hurry …)